mlir/lib/Conversion/MIOpenToGPU/MIOpenToGPU.cpp

```diff
@@ -125,12 +125,53 @@ struct MIIdRewritePattern : public OpRewritePattern<Tmi> {
 struct MIMFMARewritePattern : public OpRewritePattern<miopen::MFMAV2Op> {
   using OpRewritePattern<miopen::MFMAV2Op>::OpRewritePattern;
 
+  // The below implements the concatenation of 4 i8 to an i32:
+  // a[0] | (a[1] << 8) | (a[2] << 16) | (a[3] << 24)
+  void vector4i8Toi32(const Value &vec4i8, Value &i32vals,
```

Comment:
Can this be a [wrapped in a] more general utility that goes from `vector<4Nxi8>` to `vector<Nxi32>` with the special case for N=1? This function feels quite fragile.

Reply:
Are you talking about adding N as a argument? 

I feel like instead I can also parse the type of the incoming `vec4i8`, make sure it is 1d shape and dimension % 4 == 0, then use the dimension as-is. Hopefully it makes it more robust and generalized.

Reply:
I meant parsing the shape, yeah
That way, if I give the function a vector<8xi8>, I'll get a vector<2xi32> back, if I pass in vector<4xi8> I get an i32, and so on

(and if I send in a vector<3xi8> I get errors)

Reply:
Hmm I'm not sure there's a need for that because we don't have an xdlops related test cast that will treat vector<8xi8> as vector<2xi32> (it feels wrong since if they are already using vector types, why not use the right ones?). Besides, the vector to vector conversion feels like it really needs to sit in vector dialect, or at least serve as an extension of an existing op rather than in our lowering pattern. 

Right now I'd prefer an explicit assert on size equal to 4 and implement this only when it turn out to be necessary.

Reply:
Makes sense. Asserting that we're specifically converting a vector<4xi8> to an i32 is a perfectly solid way to handle things for now and we can add more generality when we need it.